float8_experimental/float8_linear.py

```diff
@@ -71,6 +71,54 @@ def _maybe_initialize_amaxes_scales_for_float8_cast(
         scale.copy_(new_scale)
 
 
+# this code was resurrected from https://github.com/pytorch-labs/float8_experimental/pull/128/files
```

Comment:
nit: Does the structure work out to put this in float8 ops?

Reply:
in how things look after this PR it would make sense, but might be good to see how the code looks after we add different granularities and the if/else branches on when to convert to lower precision.  Maybe we can revisit then?